pkg/buildah/imageCrdBuilder.go

```diff
@@ -0,0 +1,310 @@
+// Copyright © 2022 sealos.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buildah
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/util/homedir"
+
+	"github.com/openconfig/gnmi/errlist"
+
+	"github.com/containers/buildah"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	imagev1 "github.com/labring/sealos/controllers/imagehub/api/v1"
+	"github.com/labring/sealos/pkg/client-go/kubernetes"
+	"github.com/labring/sealos/pkg/utils/file"
+	"github.com/labring/sealos/pkg/utils/logger"
+)
+
+const (
+	AppPath          = "app"
+	READMEpath       = "README.md"
+	ConfigFileName   = "config.yaml"
+	AuthjsonPath     = "containers/auth.json"
+	SealosPath       = ".sealos"
+	ImagehubGroup    = "imagehub.sealos.io"
+	ImagehubVersion  = "v1"
+	ImagehubResource = "images"
+	ImagehubKind     = "Image"
+	XdgRuntimeDir    = "XDG_RUNTIME_DIR"
+)
+
+const (
+	//Read Content Output
+	ReadOrBuildAppConfigOutput = "Success get appConfig"
+	ReadImageContentOutput     = "Success read image content"
+	ReadReadmeOutput           = "Success read readme file"
+)
+
+type ImageCRDBuilder struct {
+	imagename      string
+	repositoryname string
+	clustername    string
+	username       string
+	userconfig     string
+	AppConfig      *imagev1.Image
+}
+
+type dockerAuthConfig struct {
+	Auth          string `json:"auth,omitempty"`
+	IdentityToken string `json:"identitytoken,omitempty"`
+}
+
+type dockerConfigFile struct {
+	AuthConfigs map[string]dockerAuthConfig `json:"auths"`
+	CredHelpers map[string]string           `json:"credHelpers,omitempty"`
+}
+
+func NewAndRunImageCRDBuilder(args []string) {
+	//run without error returns
+	icb := &ImageCRDBuilder{args[0], strings.Split(args[0], "/")[0], "", "", "", &imagev1.Image{}}
```

Comment:
What if image name not contains "/"

Reply:
if user don't specified image repo,eee
what about use hub.sealos.cn as default?

Reply:
Do not split and parse image names by hand ,using @lingdie's interface function.